### PR TITLE
[search] More synonyms.

### PR DESCRIPTION
**Pull request by mpimenov**
https://jira.mail.ru/browse/MAPSME-3229

### CODE REVIEW

**Comment by ygorshenin** [comment]
Tests?

**Comment by mpimenov** [comment]
PTAL

File: search/search_integration_tests/processor_test.cpp [hunk 1]
```diff
@@ -788,5 +788,29 @@ UNIT_CLASS_TEST(ProcessorTest, FuzzyMatch)
...
+  BuildWorld([&](TestMwmBuilder & builder) {
+    builder.Add(country);
+    builder.Add(city);
+  });
+
+  auto id = BuildCountry(countryName, [&](TestMwmBuilder & builder) { builder.Add(nightclub); });
+
+  {
+    TRules rules = {ExactMatch(id, nightclub)};
+    TEST(ResultsMatch("nightclub", "en", rules), ());
```
Comment (ygorshenin) [inline search/search_integration_tests/processor_test.cpp]: Please, add a test for no-results case.

Comment (mpimenov) [inline search/search_integration_tests/processor_test.cpp]: done

**Comment by mpimenov** [comment]
PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**